src/utils/device/device_buffer.cpp

```
@@ -0,0 +1,57 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "device/device_buffer.h"
+
+#include <cuda_runtime.h>
+
+#include "common/nixl_log.h"
+
+nixl_status_t
+nixlDeviceBufferAllocate(void **ptr, size_t size) noexcept {
```

Cppcheck (2.21.0)

[style] The function 'nixlDeviceBufferAllocate' is never used.

(unusedFunction)